lib/value/src/value.rs

```diff
@@ -769,6 +770,32 @@ impl Value {
         }
     }
 
+    /// Returns a reference to a field value specified by a path iter.
+    pub fn get_by_path_v2<'a>(&self, path: impl Path<'a>) -> Option<&Self> {
```

Comment:
Also, please take a look at the methods in `value/lib/kind`, as those will also need to get `v2` variants, since the `value` crate specifically exists to interop between VRL and Vector, and we need the `kind` module to also work with `lookup_v2`.

Alternatively, I'm also fine with having `Into`/`From` impls to convert between the two, but that means we'll have to do the work further down the line, as it'll keep the bad performance characteristics of `LookupBuf`.

Reply:
I'd prefer to leave this as future work. This PR was specifically for replacing `PathIter`, and it's already large enough as is. Future work is tracked here: https://github.com/vectordotdev/vector/issues/7070

Reply:
Hmm, I'm not a fan of only partially upgrading this crate, as it was specifically introduced as a need to interop between Vector and VRL when it comes to event schemas.

Having said that, I realize that technically this PR doesn't "upgrade" this crate, it merely adds new functionality to one part of the crate, but not the other.

I can relate to attempting to keep the scope of this PR contained, so I'll instead abandon my rebase on top of this work for now, and keep using lookup "v1" for now, until that future work is done.